### PR TITLE
[request-review-before-merge] Patch 2: Add optional review_team to repo_config

### PR DESCRIPTION
**Pull request by subsetpark**
## Patch 2: Add optional review_team to repo_config

- Add `review_team : string option` to repo_config.t (repo_config.ml:8) and to the .mli type.
- In the JSON parse path (repo_config.ml ~139), read the optional top-level "review_team" string, defaulting to None when absent; preserve the existing unknown-key-ignoring behavior.
- Set review_team = None in the `empty` value so a missing config file yields the feature-disabled default.
- Do not consume the field yet.

## Changes
- Add `review_team : string option` to repo_config.t (repo_config.ml:8) and to the .mli type.
- In the JSON parse path (repo_config.ml ~139), read the optional top-level "review_team" string, defaulting to None when absent; preserve the existing unknown-key-ignoring behavior.
- Set review_team = None in the `empty` value so a missing config file yields the feature-disabled default.
- Do not consume the field yet.

## Files to Modify
- lib_core/repo_config.ml (modify): Add review_team field to t and parse the optional "review_team" JSON string.
- lib_core/repo_config.mli (modify): Expose review_team in the public type.

## Gameplan Specification

```
module REQUEST_REVIEW.

> ══════════════════════════════════════════
> THE GUARANTEE
> ══════════════════════════════════════════
> When a PR meets every merge criterion except human approval, and the
> repo configures a review team, onton requests that team's review —
> exactly once per head commit. Approval remains GitHub's existing merge
> gate (merge_ready already blocks on REVIEW_REQUIRED); with dismiss-on-
> push enabled, a new push reopens both the gate and a fresh request.

Patch.
Branch.
Oid.
Config.
ReviewDecision.
review-required => ReviewDecision.
approved => ReviewDecision.
changes-requested => ReviewDecision.
no-review => ReviewDecision.
main => Branch.

has-pr? p: Patch => Bool.
mergeable? p: Patch => Bool.
checks-passing? p: Patch => Bool.
unresolved-comments p: Patch => Nat0.
draft? p: Patch => Bool.
busy? p: Patch => Bool.
needs-intervention? p: Patch => Bool.
base-branch p: Patch => Branch.
review-decision p: Patch => ReviewDecision.
head-oid p: Patch => Oid.
review-requested-for-oid p: Patch => Oid.
review-request-inflight? p: Patch => Bool.
should-request-review? p: Patch => Bool.
review-team-set? c: Config => Bool.
requested? p: Patch, c: Config => Bool.
---
> Characterization of the request-due state.
all p: Patch | should-request-review? p <-> (has-pr? p and mergeable? p and checks-passing? p and unresolved-comments p = 0 and ~(draft? p) and ~(busy? p) and ~(needs-intervention? p) and base-branch p = main and review-decision p = review-required and head-oid p ~= review-requested-for-oid p and ~(review-request-inflight? p)).

> Requests fire only with a configured team.
all p: Patch, c: Config | (should-request-review? p and review-team-set? c) -> requested? p c.
all p: Patch, c: Config | ~(review-team-set? c) -> ~(requested? p c).

> At most once per head commit.
all p: Patch, c: Config | requested? p c -> review-requested-for-oid p = head-oid p.

```

## Patch Specification

```
module REQUEST_REVIEW_PATCH_2.

> Patch 2 adds the optional review-team repo configuration field.
> Absent field means the feature is disabled for that repo.

Config.
Team.
review-team c: Config => Team.
review-team-set? c: Config => Bool.
---
true.

```

## Implementation Notes

- I kept the change strictly additive: `review_team` is parsed only in `Repo_config.parse_string` and defaults to `None` in `empty`, so older config files and missing files continue to load unchanged.
- The parser follows the existing top-level-object/unknown-key behavior. One small choice: an explicit empty string is normalized to `None` rather than preserved as a slug, matching the repo’s other “blank means unset” config patterns.
- The new field is intentionally not consumed anywhere yet. Downstream patches can treat `Repo_config.review_team` as the sole config toggle, with `None` meaning the feature is off for that repo.

- Spec/Evidence Mapping:
  - `Patch 2 adds the optional review-team repo configuration field.` -> `lib_core/repo_config.ml` / `lib_core/repo_config.mli` record extension and `parse_string` field decode.
  - `Absent field means the feature is disabled for that repo.` -> `Repo_config.empty.review_team = None` plus `load` returning `empty` when `config.json` is absent.
  - Consulted existing `config-loading` path (`lib_core/repo_config.ml`, `lib_core/repo_config.mli`, `bin/main.ml`) before editing to preserve the current config-loading contract.
  - Verified with `dune build` and `dune runtest`; added parser tests for absent and present `review_team`.

## Summary by CodeRabbit

* **New Features**
  * Added an optional repository-level `review_team` configuration value. Provide a team string to enable the “review-request” behavior.
  * Leaving `review_team` empty (including whitespace-only) disables the feature.
* **Bug Fixes**
  * Improved configuration parsing to reject invalid non-string `review_team` values with clear errors.

### CODE REVIEW

File: lib_core/repo_config.ml [hunk 1]
```diff
@@ -317,6 +329,20 @@ let%test "parse_string: reviewBackends absent -> empty list" =
...
+  | Ok t -> Option.is_none t.review_team
+  | Error _ -> false
+
+let%test "parse_string: review_team present -> parsed" =
+  let raw =
+    {|{"review_team":"platform","routing":{"1":{"backend":"claude"}}}|}
+  in
+  match parse_string ~known_backends:[ "claude" ] raw with
+  | Ok t -> Option.equal String.equal t.review_team (Some "platform")
+  | Error _ -> false
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply lib_core/repo_config.ml:344]: Added both missing boundary tests: whitespace-only `review_team` normalizes to `None`, and non-string input now has a dedicated rejection test.
